Gems/DebugDraw/Assets/Shaders/ObbIntersection.azsl

```diff
@@ -54,6 +55,8 @@ bool IntersectRayAABB(float3 rayOrigin, float3 rayDirection, float3 aabbMin, flo
             {
                 return false;
             }
+
+            uv[uvAxis++] = (coord - aabbMin[i]) / (aabbMax[i] - aabbMin[i]);
```

Comment:
uv is float2 but here you are accessing it as an array that assumes 3 elements. Is this correct?

Reply:
Line 59 will only be reached at max two times, since the if condition `if (whichPlane != i)` will be false once.